scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -751,10 +751,19 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       leftTok2tok(matching(token)) match {
         case FormatToken(RightParenOrBracket(), l @ LeftParen(), _) =>
           loop(l)
-        // This case only applies to classes
-        case f @ FormatToken(RightBracket(), mod, _) if mod.is[Modifier] =>
-          loop(next(f).right)
-        case FormatToken(r @ RightParenOrBracket(), _, _) => Some(r)
+        case f @ FormatToken(left @ RightParenOrBracket(), right, _) =>
+          right match {
+            case Modifier() =>
+              // This case only applies to classes
+              next(f).right match {
+                case x @ (_: Token.LeftParen | _: Token.LeftBracket) =>
```

Comment:
The fix is ugly but it seems to do the job, feel free to refactor @pjrt I'm not sure exactly what this case is about

Reply:
Unn, I haven't looked at this code in a while. But it doesn't look any less ugly than the other stuff we need to do in this file :).